src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt

```diff
@@ -41,25 +49,18 @@ class CommandsTabPanel(
 
     if (ConfigUtil.isFeatureFlagEnabled("cody.feature.inline-edits") ||
         CodyApplicationSettings.instance.isInlineEditionEnabled) {
-      addInlineEditActionButton("cody.editCodeAction")
-      addInlineEditActionButton("cody.documentCodeAction")
-      addInlineEditActionButton("cody.testCodeAction")
+      addInlineEditActionButton(EditCodeAction.ID)
+      addInlineEditActionButton(DocumentCodeAction.ID)
+      addInlineEditActionButton(TestCodeAction.ID)
     }
 
-    addHierarchyListener {
-      if (!project.isDisposed) {
-        if (it.component.isShowing) {
-          IgnoreOracle.getInstance(project).addListener(this)
-        } else {
-          IgnoreOracle.getInstance(project).removeListener(this)
-        }
-      }
-    }
+    IgnoreOracle.getInstance(project).addListener(this)
```

Comment:
@dominiccooney I simplified this, I do not see any reason to keep adding and removing this listener.
If there is any, please let me know.

Reply:
Should be fine... I was just keen to avoid references from something held on to by agent callbacks *and* the opaque JetBrains machinery. Seems like a recipe for leaks.